### PR TITLE
Adding supply warning for non-update-to-date pairs

### PR DESCRIPTION
**Pull request by IanWoodard**
Added a banner warning the user that a pair has not been updated in the last two weeks rather than hiding it.
image

### CODE REVIEW

File: earn/src/components/markets/modal/SupplyModal.tsx [hunk 1]
```diff
@@ -67,6 +76,8 @@ function getConfirmButton(state: ConfirmButtonState, token: Token): { text: stri
       return { text: 'Check Wallet', enabled: false };
     case ConfirmButtonState.READY:
       return { text: 'Confirm', enabled: true };
+    case ConfirmButtonState.WAITING_FOR_CONFIRMATION:
+      return { text: 'Please Confirm', enabled: false };
```
Comment (haydenshively) [inline earn/src/components/markets/modal/SupplyModal.tsx:80]: Seems unused? Also, I believe we say "Check wallet" as the button text for this type of thing in other places.

File: earn/src/components/markets/modal/SupplyModal.tsx [hunk 2]
```diff
@@ -244,10 +255,21 @@ export default function SupplyModal(props: SupplyModalProps) {
...
-    <Modal isOpen={isOpen} setIsOpen={setIsOpen} title='Supply'>
+    <Modal isOpen={isOpen} setIsOpen={setIsOpen} title='Supply' maxHeight='650px'>
       <div className='w-full flex flex-col gap-4'>
+        {!wasUpdatedInPast2Weeks && (
+          <div className='border-2 border-caution flex items-center p-2 gap-2 rounded-lg'>
+            <AlertTriangleWrapper>
+              <AlertTriangle width={24} height={24} />
+            </AlertTriangleWrapper>
+            <Text size='XS' className='w-full'>
+              This market hasn't been updated in the past 2 weeks.
```
Comment (haydenshively) [inline earn/src/components/markets/modal/SupplyModal.tsx:269]:
```suggestion
              In the past two weeks, no one has updated the implied volatility (IV) for this market. IV impacts risk parameters like LLTV. It's normally updated multiple times per day.
```

File: shared/src/util/Numbers.ts [hunk 3]
```diff
@@ -177,25 +177,34 @@ export function formatTokenAmountCompact(amount: number, length = 4): string {
...
+    return (
+      sign +
+      amount.toLocaleString('en-US', {
+        style: 'decimal',
+        notation: 'compact',
+        compactDisplay: 'short',
+        maximumSignificantDigits: length,
+        minimumSignificantDigits: 2,
+      })
+    );
```
Comment (haydenshively) [inline shared/src/util/Numbers.ts:189]: Something up with your linter? Seems to be going back and forth on a few things like this.

File: earn/src/components/markets/modal/SupplyModal.tsx (hunk 1 above)
Comment (IanWoodard) [reply earn/src/components/markets/modal/SupplyModal.tsx:80]: Yep, left over from an earlier implementation that required confirming explicitly

File: shared/src/util/Numbers.ts (hunk 3 above)
Comment (IanWoodard) [reply shared/src/util/Numbers.ts:189]: It seems to like it like this now, will keep an eye on it.

**Pull request merged**